neuro_extras/main.py

```diff
@@ -355,99 +363,48 @@ async def _data_cp(
 
     else:
```

Comment:
Ohh this method is over-complicated... it needs more system, at least in var naming. This PR makes a right step extracting the logic of compress/extract to separate methods, we'll need to go on with the refactoring.